pkg/utils/provider_meta.go

```diff
@@ -0,0 +1,94 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/MaterializeInc/terraform-provider-materialize/pkg/clients"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmoiron/sqlx"
+)
+
+type ProviderMeta struct {
+	DB             map[clients.Region]*clients.DBClient
+	Frontegg       *clients.FronteggClient
+	CloudAPI       *clients.CloudAPIClient
+	DefaultRegion  clients.Region
+	RegionsEnabled map[clients.Region]bool
+}
+
+var DefaultRegion string
+
+func GetProviderMeta(meta interface{}) (*ProviderMeta, bool) {
+	providerMeta, ok := meta.(*ProviderMeta)
+	if !ok || providerMeta == nil {
+		fmt.Println("Type assertion failed: provider meta is not of type *ProviderMeta or is nil")
```

Comment:
Should this be a panic?

Reply:
Agreed; seems like this could just be `providerMetda := meta.(*ProviderMeta)`.

Reply:
Just updated this accordingly, now you can use this similar to other functions in the resources and data sources, eg:

```golang
providerMeta, err := utils.GetProviderMeta(meta)
if err != nil {
	return diag.FromErr(err)
}
```